syntax = "proto3";

option java_package = "com.google.android.apps.aiy.proto";

// Messages sent from phone to AIY.
message Request {
  oneof request {
    GetState.Request get_state = 1;
    Identify.Request identify = 2;
    ScanNetworks.Request scan_networks = 3;
    ConnectNetwork.Request connect_network = 4;
    SetDeviceName.Request set_device_name = 5;
  }
}

// Messages sent from AIY to phone.
message Response {
  oneof response {
    GetState.Response get_state = 1;
    Identify.Response identify = 2;
    ScanNetworks.Response scan_networks = 3;
    ConnectNetwork.Response connect_network = 4;
    SetDeviceName.Response set_device_name = 5;
  }
}

enum ResponseStatus {
  UNKNOWN = 0;
  SUCCESS = 1;
  FAILURE = 2;
}

message WifiInfo {
  enum WifiState {
    UNKNOWN = 0;
    DISABLED = 1;
    DISCONNECTED = 2;
    CONNECTED = 3;
  }
  WifiState state = 1;
  string ssid = 2;
  string ip = 3;
  sint32 rssi = 4;
}

message GetState {
  message Request {
  }

  message Response {
    ResponseStatus status = 1;
    uint32 protocol_version = 2;
    WifiInfo wifi_info = 3;
    uint32 kit_id = 4;
    string device_name = 5;
  }
}

message Identify {
  message Request {
    string data = 1;
    bytes extra = 2;
  }

  message Response {
    ResponseStatus status = 1;
    string data = 2;
    bytes extra = 3;
  }
}


message ScanNetworks {
  message Request {
    uint32 timeout = 1;
  }

  message Response {
    ResponseStatus status = 1;
    message ScanResult {
      string ssid = 1;
      bool secure = 2;
      sint32 rssi = 3;
    }
    repeated ScanResult results = 2;
  }
}

message ConnectNetwork {
  message Request {
    uint32 timeout = 1;
    string ssid = 2;
    string psk = 3;
  }

  message Response {
    ResponseStatus status = 1;
    WifiInfo wifi_info = 2;
  }
}

message SetDeviceName {
  message Request {
    string device_name = 1;
  }

  message Response {
    ResponseStatus status = 1;
    string device_name = 2;
  }
}
